MetaMask/metamask-extension
# Rename `err` property to `error` in the transaction meta

## **Description**
 The core TransactionController uses `error` to log transaction error information and for the sake of standardization and minimizing code maintenance, `err` is renamed in this PR as part of the unification of the Transaction Controller effort.
 This migration adds an `error` property by copying the old `err` and deleting it afterwards.

## **Manual testing steps**

**Scenario 1**
- Go to Test Dapp
- In Settings enable Customize nonce
- Send a Transaction with a nonce already used
- Go to Settings > Advance > Download state logs
- Check the property "error" in the state logs for this transaction with a "nonce too low: ..."

**Scenario 2**
- Go to the test-dapp
- Create a token
- Perform a transaction

Both scenarios should be successfully completed, as usual, with no functional changes.

## **Related issues**

Fixes https://github.com/MetaMask/MetaMask-planning/issues/1470

## **Pre-merge author checklist**

- x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).

## **Pre-merge reviewer checklist**
